editor.html

```diff
@@ -261,81 +274,14 @@ <h2><i class="fa fa-unlock-alt"></i> <strong>{{ title }}</strong></h2>
-        'ei': 'else if some other condition is True, do something',
-        'el': 'else do some other thing',
-        'for': 'for each item in a collection of items do something with each item',
-        'try': 'try doing something and handle exceptions (errors)'
-      },
-      'share': {
-        'title': 'Share Code',
-        'instructions': 'Use a password and optional hint (to help you remember the password) to securely create a link to share your code with others.',
-        'passphrase': 'Password: ',
-        'hint': 'Password hint: ',
-        'button': 'Create Link',
-        'description': 'This URL points to your code:',
-        'shortener': 'This is a short version of the link:'
-      },
-      'decrypt': {
-        'title': 'Decrypt Source Code',
-        'instructions': 'Enter a password to decrypt the source code.',
-        'passphrase': 'Password: ',
-        'button': 'Decrypt'
-      },
-      'alerts': {
-        'download': 'Safari has a bug that means your work will be downloaded as an un-named file. Please rename it to something ending in .hex. Alternatively, use a browser such as Firefox or Chrome. They do not suffer from this bug.',
-        'save': 'Safari has a bug that means your work will be downloaded as an un-named file. Please rename it to something ending in .py. Alternatively, use a browser such as Firefox or Chrome. They do not suffer from this bug.',
-        'length': 'Oops! Your script is too long given the limited memory on the device.',
-        'unrecognised_hex': "Sorry, we couldn't recognise this file",
-        'snippets': 'Snippets are disabled when blockly is enabled.'
-      },
-      'confirms': {
-        'quit': "Some of your changes have not been saved. Quit anyway?",
-        'blocks': "You have unsaved code. Using blocks will change your code. You may lose your changes. Do you want to continue?"
-      },
-      'code': {
-        'start': "# Add your Python code here. E.g.\n" +
-                 "from microbit import *\n\n\n" +
-                 "while True:\n" +
-                 "    display.scroll('Hello, World!')\n" +
-                 "    display.show(Image.HEART)\n" +
-                 "    sleep(2000)\n"
-      },
-      'drop': {
-        'python': 'Extracted from a Python file',
-        'hex': 'Extracted from a hex file'
-      },
-      'load': {
-        'title': 'Load Script or HEX file',
-        'instructions': 'Drag and drop a .hex or .py file in here to open it.',
-        'submit': 'Load',
-        'toggle-drop': 'Or drag and drop a file',
-        'toggle-file': 'Or pick a file'
-      }
-    },
-    'flags': {
-      'blocks': false,
-      'snippets': true,
-      'experimental': false,
-      'share': false
-    },
-});
-        </script>
+        <script src="static/js/translation.js"></script>
+  <script>
+      // Call the web_editor function to start the editor running.
+          document.getElementById("lang").onload = function(){
```

Comment:
Does this id ("lang") exists in the HTML dom at this point?
It's possible this element is only created by `static/js/translation.js`?
In that case, it's possible if that js file is not loaded and execute before this line is executed, and that point it won't be able to find an element to attache this event listener.
You can do a general on page load, should be more robust.

Reply:
Ahh okay, are you suggesting that the page should load regardless of the language script in case it doesn't load at all or in time? (in other words so that the rest of the page is not dependent on the language loading or failing to load)

Reply:
My concern is: when does an element with the id "lang" is first added to the DOM? Does the HTML in the editor.html file contain an element with that ID? Is the element dynamically created and attached to the DOM via the `static/js/translation.js` file?

If added in the translations.js file, then it's not a guarantee that by the time the line `document.getElementById("lang").onload = function()...` is executed that the element will exist already. This is because the browser can start downloading the `static/js/translation.js` file, and then continue executing whatever is below before it had a chance to finish downloading it and executing it. In that case then `document.getElementById("lang")` would be equal to `null` and the onload event will never be registered.

Reply:
I've implemented the window.onload now so it no longer waits for the "lang" element. 

I see what you mean that the lang element may not even exist let alone be loading when the browser reaches the lang.onload. 

How does window.onload guarantee that the language js file is loaded? From my understanding it waits until all the html is loaded, however this would also mean that if the script tag "lang" isn't added in time it will result in the language variable coming up as null.